src/libsystemd/sd-bus/bus-dump.c

```diff
@@ -591,5 +636,13 @@ int bus_message_pcap_frame(sd_bus_message *m, size_t snaplen, FILE *f) {
                 snaplen -= w;
         }
 
+        if (padlen > 0) {
+                const uint32_t zero = 0;
+                fwrite(&zero, 1, padlen, f);
```

Comment:
Please add `assert(padlen <= sizeof(uint32_t));`
Also, `fwrite(&(uint32_t) {}, 1, padlen, f);`?

Reply:
reworked this in later  code version